### PR TITLE
Stream Datastore Dump in Flask View

### PR DESCRIPTION
**Pull request by JVickery-TBS**
help me

### CODE REVIEW

File: ckanext/datastore/blueprint.py [hunk 1]
```diff
@@ -25,6 +27,10 @@
     json_writer,
     xml_writer,
 )
+if six.PY2:
+    from StringIO import StringIO
```
Comment (wardi) [inline ckanext/datastore/blueprint.py:31]:
```suggestion
    from cStringIO import StringIO
```

File: ckanext/datastore/blueprint.py [hunk 2]
```diff
@@ -64,30 +70,109 @@ def dump(resource_id):
...
-                if k in [
-                    u'filters', u'q', u'distinct', u'plain', u'language',
-                    u'fields'
-                ]
-            },
+    if fmt == u'csv':
+        writer_factory = csv_writer
+        records_format = u'csv'
+        content_disposition = u'attachment; filename="{name}.csv"'.format(
+                                    name=encode_rfc2231(resource_id))
```
Comment (wardi) [inline ckanext/datastore/blueprint.py:91]: it looks like this does nothing without language/charset set? https://docs.python.org/3/library/email.utils.html#email.utils.encode_rfc2231

**Comment by wardi** [comment]
excluding the test changes would make merging upstream ckan fixes cleaner, but otherwise LGTM

**Comment by JVickery-TBS** [comment]
@JVickery-TBS okay the test changes have been reverted back to `canada-v2.9` so no changes on them.

**Review by wardi — APPROVED** [review]

**Pull request merged**